pkg/network/server.go

```diff
@@ -404,6 +423,26 @@ func (s *Server) handleBlockCmd(p Peer, block *core.Block) error {
 	return s.bQueue.putBlock(block)
 }
 
+// handlePing processes ping request.
+func (s *Server) handlePing(p Peer, ping *payload.Ping) error {
+	return p.WriteMsg(NewMessage(s.Net, CMDPong, payload.NewPing(s.id, s.chain.BlockHeight())))
+}
+
+// handlePing processes pong request.
+func (s *Server) handlePong(p Peer, pong *payload.Ping) error {
+	if !p.PingSent() {
+		errorMsg := "pong message wasn't expected"
+		s.log.Error(errorMsg)
+		return errors.New(errorMsg)
+	}
+	p.UpdatePingSent(false)
+	if s.chain.HeaderHeight() < pong.LastBlockIndex {
+		p.UpdateLastBlockIndex(pong.LastBlockIndex)
```

Comment:
This update should be done unconditionally, not depending on relation to `HeaderHeight()`.